corehq/apps/reports/tests/test_views.py

```diff
@@ -0,0 +1,578 @@
+        url = f'{base_url}?{params}'
+        request = RequestFactory().get(url)
+        request.user = self.user
+        return request
+
+    def _set_user_report_access(self, *report_names):
+        self.reports_role.permissions = Permissions(view_report_list=list(report_names))
+        self.reports_role.save()
+
+
+class TestDeleteConfig(TestCase):
+    def test_invalid_config_id_returns_404(self):
+        with self.assertRaises(Http404):
+            self.delete_config(config_id='nonexistant_config_id')
+
+    def test_mismatched_domains_returns_404(self):
+        report = self._create_saved_report(domain='not_your_domain')
+
+        with self.assertRaises(Http404):
+            self.delete_config(domain=self.domain, config_id=report._id)
+
+    def test_non_owner_cannot_delete_saved_report(self):
+        report = self._create_saved_report(user_id='not_you')
+
+        with self.assertRaises(Http404):
+            self.delete_config(config_id=report._id)
+
+    def test_can_delete_saved_report(self):
+        report = self._create_saved_report(domain=self.domain, user_id=self.user._id)
+
+        response = self.delete_config(self.domain, report._id)
+        self.assertEqual(response.status_code, 200)
+
+        with self.assertRaises(ResourceNotFound):
+            ReportConfig.get(report._id)
+
+# ############ Helper / Setup
+    def delete_config(self, domain=None, config_id='test'):
+        domain = domain or self.domain
+        return views.delete_config(self.request, domain, config_id)
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'test-domain'
+        cls.domain_obj = create_domain(cls.domain)
+        cls.user = WebUser.create(cls.domain, 'test_user', 'test_password', None, None)
+        cls.user.is_authenticated = True
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete(deleted_by=None)
+        cls.domain_obj.delete()
+        super().tearDownClass()
+
+    def setUp(self):
+        super().setUp()
+        self.request = self._create_request()
+
+    def _create_request(self):
+        url = f'/a/{self.domain}/deleteConfig'
+        request = RequestFactory().delete(url)
+        request.user = self.user
+        return request
+
+    def _create_saved_report(self, domain=None, user_id=None):
+        config = ReportConfig(
+            domain=domain or self.domain,
+            owner_id=user_id or self.user._id,
+            name='Test',
+            description='Test Saved Report',
+            report_slug='worker_activity',
+            report_type='project_report'
+        )
+        config.save()
+        self.addCleanup(config.delete)
+
+        return config
+
+
+class TestDeleteScheduledReport(TestCase):
+    def test_invalid_report_redirects_to_home(self):
+        response = self.delete_scheduled_report(report_id='invalid_report_id')
+        self.assertEqual(response.status_code, 302)
```

Comment:
Should this test be renamed since it does not check that it redirects _to home_?
```suggestion
    def test_invalid_report_redirects(self):
        response = self.delete_scheduled_report(report_id='invalid_report_id')
        self.assertEqual(response.status_code, 302)
```
Alternately, maybe there should be an assertion about the redirected URL?